src/ArrayList.java

```diff
@@ -0,0 +1,100 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+class ArrayList<T> implements ListForTask<T> {
+	private static final int CAPACITY_OF_ARRAY = 10;
+	public T[] newOwnArray;
+	private int arraySize;
+
+	public ArrayList() {
+		this.newOwnArray = (T[]) new Object[CAPACITY_OF_ARRAY];
+	}
+
+	@Override
+	public void add(T value) {
+		if (arraySize >= newOwnArray.length - 1) {
+			newOwnArray = Arrays.copyOf(newOwnArray, newOwnArray.length + (newOwnArray.length >> 1));
+		}
+		newOwnArray[arraySize] = value;
```

Comment:
you can write it more simple:
newOwnArray[arraySize] = value;
		arraySize++;
   ->
newOwnArray[arraySize++] = value;